src/main-process/atom-application.js

```diff
@@ -75,18 +76,26 @@ const getExistingSocketSecret = (atomVersion) => {
   return fs.readFileSync(socketSecretPath, 'utf8')
 }
 
-const createSocketSecret = (atomVersion) => {
-  const socketSecret = crypto.randomBytes(16).toString('hex')
+const getRandomBytes = promisify(crypto.randomBytes)
+const writeFile = promisify(fs.writeFile)
 
-  fs.writeFileSync(getSocketSecretPath(atomVersion), socketSecret, {encoding: 'utf8', mode: 0o600})
+const createSocketSecret = async (atomVersion) => {
+  const socketSecret = (await getRandomBytes(16)).toString('hex')
+
+  await writeFile(getSocketSecretPath(atomVersion), socketSecret, {encoding: 'utf8', mode: 0o600})
 
   return socketSecret
 }
 
 const encryptOptions = (options, secret) => {
   const message = JSON.stringify(options)
 
-  const initVector = crypto.randomBytes(16)
+  // Even if the following IV is not cryptographically secure, there's a really good chance
+  // it's going to be unique between executions which is the requirement for GCM.
```

Comment:
I think it might be helpful to future readers if you explain the motivation for avoiding cryptographic randomness on this code path (that it's too slow).

Reply:
I've added more explicit comments for both things in https://github.com/atom/atom/pull/19246